include/merlin/core_kernels/upsert_and_evict.cuh

```diff
@@ -21,6 +21,1320 @@
 namespace nv {
 namespace merlin {
 
+// Use 1 thread to deal with a KV-pair, including copying value.
+template <typename K = uint64_t, typename V = byte4, typename S = uint64_t,
+          typename VecV = byte16, uint32_t BLOCK_SIZE = 128>
+__global__ void tlp_v1_upsert_and_evict_kernel_unique(
+    Bucket<K, V, S>* __restrict__ buckets, int32_t* __restrict__ buckets_size,
+    const uint64_t buckets_num, uint32_t bucket_capacity, const uint32_t dim,
+    const K* __restrict__ keys, const VecV* __restrict__ values,
+    const S* __restrict__ scores, K* __restrict__ evicted_keys,
+    VecV* __restrict__ evicted_values, S* __restrict__ evicted_scores,
+    uint64_t n, uint64_t* __restrict__ evicted_counter) {
+  using BUCKET = Bucket<K, V, S>;
+  using CopyValue = CopyValueMultipleGroup<VecV, 1>;
+
+  // bucket_capacity is a multiple of 4.
+  constexpr uint32_t STRIDE_S = 4;
+  constexpr uint32_t Load_LEN_S = sizeof(byte16) / sizeof(S);
+  __shared__ S sm_bucket_scores[BLOCK_SIZE][2 * STRIDE_S];
+
+  uint32_t tx = threadIdx.x;
+  uint32_t kv_idx = blockIdx.x * blockDim.x + tx;
+  K key{static_cast<K>(EMPTY_KEY)};
+  S score{static_cast<S>(EMPTY_SCORE)};
+  OccupyResult occupy_result{OccupyResult::INITIAL};
+  VecD_Comp target_digests{0};
+  VecV* bucket_values_ptr{nullptr};
+  K* bucket_keys_ptr{nullptr};
+  int32_t* bucket_size_ptr{nullptr};
+  uint32_t key_pos = {0};
+  uint32_t evict_idx{0};
+  uint32_t bucket_size{0};
+  if (kv_idx < n) {
+    key = keys[kv_idx];
+    score = scores != nullptr ? scores[kv_idx] : static_cast<S>(MAX_SCORE);
```

Comment:
A quick question: this score would be treated as the `desired score`, right?

Reply:
If user provided score, it will be `desired score`;
But if not, the `score` will be the `MAX_SCORE`, representing its hign priority. But when insert into bucket, it will be update by device clock for LRU mode.